TeamCode/src/main/java/org/firstinspires/ftc/teamcode/LinearOp.java

```diff
@@ -56,48 +59,102 @@
 @TeleOp(name="Basic: Linear OpMode", group="Linear Opmode")
 public class LinearOp extends LinearOpMode {
 
+    //final private Servo S1 = hardwareMap.get(Servo.class, Statics.Servos.left_glyphGrabber);
+
+    //Initialize objects
+    private MecanumDrive mWheel;
+    private ServoControl jArm;
+    private GamepadSpace previous;
+
     // Declare OpMode members.
-    private ElapsedTime runtime = new ElapsedTime();
+    final private ElapsedTime runtime = new ElapsedTime();
+    private boolean isDPadUpChanged = false;
+    private boolean isDPadDownChanged = false;
+    private boolean isJleftXChanged = false;
+    private boolean isJleftYChanged = false;
+    private boolean isJrightXChanged = false;
+    private boolean isLBChanged = false;
+
+
+    private void detectGPChange() {
+        isDPadUpChanged   = gamepad1.dpad_up != previous.DPadUp;
+        isDPadDownChanged = gamepad1.dpad_down != previous.DPadDown;
+        isLBChanged       = gamepad1.left_bumper != previous.LB;
+        isJleftXChanged   = gamepad1.left_stick_x != previous.JleftX;
+        isJleftYChanged   = gamepad1.left_stick_y != previous.JleftY;
+        isJrightXChanged  = gamepad1.right_stick_x != previous.JrightX;
+        boolean isLTChanged = gamepad1.left_trigger != previous.LT;
+        boolean isRTChanged = gamepad1.right_trigger != previous.RT;
```

Comment:
the last 2 lines have no effect

Reply:
I know as long as Android Studio already told me when I was writing it. This is preserved for next version of code, which adds control code for Glyph Grabber arm and the lifter. (They prefer to use triggers ._.)